src/main/java/org/testobject/appium/junit/TestObjectAppiumSuite.java

```diff
@@ -211,20 +200,26 @@ private static TestObject getConfig(Class<?> clazz) {
 		return testobject;
 	}
 
-	private Set<String> getRemoteDeviceIds() {
+	private Set<DataCenterSuite> getDataCenterSuites() {
 		AppiumSuiteResource suiteReportResource = new AppiumSuiteResource(client);
-
-		return suiteReportResource.readSuiteDeviceIds(testObjectSuiteId);
+		Set<DataCenterSuite> dataCenterSuites = suiteReportResource.readDataCenterSuites(testObjectSuiteId);
+		for (DataCenterSuite dataCenterSuite : dataCenterSuites) {
+			if (dataCenterSuite.getDeviceDescriptorIds().isEmpty() == false) {
+				return dataCenterSuites;
```

Comment:
I don't quite get this logic. What did you try to do here? If you are doing validation here maybe putting that into a method to explain it would be a better idea